travis.sh

```diff
@@ -4,25 +4,14 @@ set -e
 mkdir build
 cd build
 
-if [ -n "${COVERAGE}" ]; then
 
+if [ -n "${COVERAGE}" ]; then
   # build
-  cmake .. -DCMAKE_C_STANDARD=${C_STANDARD:-99} -DCMAKE_CXX_STANDARD=${CXX_STANDARD:-98} -DBUILD_CFP=ON -DZFP_WITH_OPENMP=ON -DCMAKE_C_FLAGS=-coverage -DCMAKE_CXX_FLAGS=-coverage;
-  cmake --build . ;
-  ctest -V -C "Debug";
-
+  ctest -V -C "Debug" -DC_STANDARD=${C_STANDARD:-99} -DCXX_STANDARD=${CXX_STANDARD:-98} -DBUILD_CFP=ON -DBUILD_OPENMP=ON -DBUILD_CUDA=OFF -DWITH_COVERAGE=ON -S $TRAVIS_BUILD_DIR/cmake/travis.cmake
 else
-
   # build/test without OpenMP, with CFP
-  cmake .. -DCMAKE_C_STANDARD=${C_STANDARD:-99} -DCMAKE_CXX_STANDARD=${CXX_STANDARD:-98} -DZFP_WITH_OPENMP=OFF -DBUILD_CFP=ON;
-  cmake --build . ;
-  ctest -V -C "Debug";
-
-  rm -rf ./* ;
```

Comment:
We create two builds per CI job because some tests are exclusive to builds with/without OpenMP. In this commit's travis log, it looks like the OpenMP tests are never compiled or run (named testZfpOmp*). I think the `rm -rf ./*` to clear the `build/` directory is missing.

Is CDash capable of handling multiple builds in 1 CI job? If we split this into separate jobs, then we double our CI build matrix.

Reply:
We can handle multiple builds in a single CI job. We just have to make sure we name the builders differently, otherwise they combine together in the CDash database. We shouldn't need to remove the build directory, but will need to explicitly state OpenMP should be enabled on the second invocation.

I will update the MR to include this.

Reply:
I have updated the MR to have CDash submit both for OpenMP and No-OpenMP.

Reply:
I realized some machines we get on travis do not support openmp, which is why I wasn't seeing those tests in the output. However, I still support the `rm -rf ./*` because the new CMake code will upload the 1st build's test results (without openmp) twice if the 2nd build fails. Take a look on CDash at an OSX build on the general dashboard page: we expect 148 tests to be run when openmp is enabled, but see 130 for OSX builds with openmp.

Reply:
On second thought I think the right thing to do is to conditionally build the 2nd time only if openmp is available on the machine. If a CI machine is always expected to fail the openmp build, it will only clutter the dashboard.

Reply:
Do you know of a way to figure in the the travis script if the machine will support `OpenMP`?

Reply:
I don't see any direct easy way to check through the command line. I think we could make a short helper CMake script to run `find_package(OpenMP COMPONENTS C REQUIRED)`. The travis script could conditionally start a 2nd build, depending on that helper cmake code succeeded or failed.